backend/src/main/java/com/bakdata/conquery/models/datasets/concepts/filters/postalcode/PostalCodesManager.java

```diff
@@ -0,0 +1,106 @@
+package com.bakdata.conquery.models.datasets.concepts.filters.postalcode;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.univocity.parsers.common.processor.BeanListProcessor;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class PostalCodesManager {
+
+	@Getter
+	@Setter
+	static private File dataFile = null;
+
+
+	@Getter
+	final static private HashMap<Set<String>, Double> data = new HashMap<>();
+
+	static private boolean loaded = false;
+
+
+	/**
+	 * This method loads the postcodes and their distances between each other
+	 * Then it gives a list of the entries read.
+	 *
+	 * @return The list is NULL if the file concerned has not been set yet
+	 */
+	static public List<PostalCodeDistanceEntity> loadData() throws IOException, URISyntaxException {
+		if (getDataFile() == null) {
+			setDataFile(new File(PostalCodesManager.class.getClassLoader().getResource("com.bakdata.conquery/postalcodes.csv").toURI()));
+		}
+		final BeanListProcessor<PostalCodeDistanceEntity> rowProcessor = new BeanListProcessor<>(PostalCodeDistanceEntity.class);
```

Comment:
Ich kann mir vorstellen, dass dieser Processor sehr langsam und damit ungeeignet für das ist was wir machen.
Kannst du benchmarken, wie er sich im vergleich zu records oder sogar einem eigenen processor verhält der auf die indizes geht?

Reply:
https://www.univocity.com/pages/univocity_parsers_row_processors.html#other-row-processors

Hier eine eigene Implementierung die wirklich deutlich schneller ist im Preprocessing aber einfach noch nicht geschafft wurde zu mergen: https://github.com/bakdata/conquery/pull/1445/files#diff-be16c4a311976f2361c456bd758451ce0364cddf827f756c3b9fb2fe10bd4a11R243

Reply:
wie kommst du darauf dass der Standard-Prozessor langsamer ist ??

Reply:
Das ist nicht der StandardProzessor, und weil es in deren Dokumentation so steht. Du gehst dann ja pro Zeile über einen String in eine Map, in ein Array oder ähnliches. Das erzeugt ordentlich overhead. Wenn du das schon vorlagern oder komplett ignorieren kannst, sparst du dir einiges an berechung.